pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -624,7 +627,11 @@ func (nc *Controller) doNoScheduleTaintingPass(ctx context.Context, nodeName str
 
 	// Map node's condition to Taints.
 	var taints []v1.Taint
+	nodeUninitializedConditionExists := false
 	for _, condition := range node.Status.Conditions {
+		if condition.Type == v1.NodeUninitialized {
+			nodeUninitializedConditionExists = true
+		}
```

Comment:
I would not expect to add specific logic like this inside this loop (and below in the `!nodeUninitializedConditionExists` block)

instead, come up with a way to indicate which taints in `nodeConditionToTaintKeyStatusMap` should be deleted if the condition is absent, and which shouldn't, so this loop remains generic

Reply:
I think we came up with the conclusion that this controller should not touch(add/delete) the taint if the corresponding condition is missing.

Reply:
right, but that should be implemented generically, not by inspecting this specific condition inside the loop